vector/src/main/java/im/vector/riotx/features/share/IncomingShareFragment.kt

```diff
@@ -0,0 +1,239 @@
+import android.content.ClipDescription
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.widget.SearchView
+import androidx.core.view.isVisible
+import com.airbnb.mvrx.fragmentViewModel
+import com.airbnb.mvrx.withState
+import com.kbeanie.multipicker.utils.IntentUtils
+import im.vector.matrix.android.api.session.content.ContentAttachmentData
+import im.vector.matrix.android.api.session.room.model.RoomSummary
+import im.vector.riotx.R
+import im.vector.riotx.core.di.ActiveSessionHolder
+import im.vector.riotx.core.extensions.cleanup
+import im.vector.riotx.core.extensions.configureWith
+import im.vector.riotx.core.extensions.exhaustive
+import im.vector.riotx.core.platform.VectorBaseFragment
+import im.vector.riotx.core.utils.PERMISSIONS_FOR_WRITING_FILES
+import im.vector.riotx.core.utils.PERMISSION_REQUEST_CODE_PICK_ATTACHMENT
+import im.vector.riotx.core.utils.allGranted
+import im.vector.riotx.core.utils.checkPermissions
+import im.vector.riotx.features.attachments.AttachmentsHelper
+import im.vector.riotx.features.attachments.preview.AttachmentsPreviewActivity
+import im.vector.riotx.features.attachments.preview.AttachmentsPreviewArgs
+import im.vector.riotx.features.login.LoginActivity
+import kotlinx.android.synthetic.main.fragment_incoming_share.*
+import javax.inject.Inject
+
+/**
+ * Display the list of rooms
+ * The user can select multiple rooms to send the data to
+ */
+class IncomingShareFragment @Inject constructor(
+        val incomingShareViewModelFactory: IncomingShareViewModel.Factory,
+        private val incomingShareController: IncomingShareController,
+        private val sessionHolder: ActiveSessionHolder
+) : VectorBaseFragment(), AttachmentsHelper.Callback, IncomingShareController.Callback {
+
+    private lateinit var attachmentsHelper: AttachmentsHelper
+    private val viewModel: IncomingShareViewModel by fragmentViewModel()
+
+    override fun getLayoutResId() = R.layout.fragment_incoming_share
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        // If we are not logged in, stop the sharing process and open login screen.
+        // In the future, we might want to relaunch the sharing process after login.
+        if (!sessionHolder.hasActiveSession()) {
+            startLoginActivity()
+            return
+        }
+        super.onViewCreated(view, savedInstanceState)
+        setupRecyclerView()
+        setupToolbar(incomingShareToolbar)
+        attachmentsHelper = AttachmentsHelper.create(this, this).register()
+
+        val intent = vectorBaseActivity.intent
+        if (intent?.action == Intent.ACTION_SEND || intent?.action == Intent.ACTION_SEND_MULTIPLE) {
+            var isShareManaged = attachmentsHelper.handleShareIntent(
+                    IntentUtils.getPickerIntentForSharing(intent)
```

Comment:
This seems to break the intent.resolveType() at handleShareIntent (AttachmentsHelper.kt), thus making it impossible to share multiple files by share intent. (Just passing the intent as-is allows for multiple files to be shared.) Why is this function called?

Reply:
Note sure about this code, but I fixed it be splitting logic for ACTION_SEND and ACTION_SEND_MULTIPLE